Add sample to parse KV error details (#36283)
Azure/azure-sdk-for-net: sdk/keyvault/Azure.Security.KeyVault.Secrets/samples/Sample4_ErrorResponse.md

> Note: Details of an error apart from `code` or `message` are not part of the service contract and should not be used for mission-critical decisions. The error code and message are conveniently defined as properties on `RequestFailedException`.

## Creating a SecretClient

To create a new `SecretClient` to attempt to set a secret, you need the endpoint to an Azure Key Vault and credentials.
You can use the [DefaultAzureCredential](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/identity/Azure.Identity/README.md) to try a number of common authentication methods optimized for both running as a service and development.

In the sample below, you can set `keyVaultUrl` based on an environment variable, configuration setting, or any way that works for your application.

```C# Snippet:SecretsSample1SecretClient
var client = new SecretClient(new Uri(keyVaultUrl), new DefaultAzureCredential());
```

## Parsing error details

Azure Key Vault will return an error response if you attempt to create a secret, key, or certificate that is in a deleted state.

> If the service returns an HTTP 409 (Conflict) error, you can use `RequestFailedException.GetRawResponse()` to get the raw HTTP response and parse the content for more details.

Reviewer: nit: 
```suggestion
If the service returns an HTTP 409 (Conflict) error, `RequestFailedException.GetRawResponse()` can be called to get the raw HTTP response and parse the content for more details.
```

Author: Microsoft Learn guidance is to use the informal first-person active voice.

Reviewer: Ah, good to know!
